### PR TITLE
KAFKA-7833: Add missing test

### PR DESCRIPTION
**Pull request by mjsax**
Quick follow up to #8825 

Call for review @guozhangwang @vvcephei

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java [hunk 1]
```diff
@@ -680,8 +680,11 @@ private void validateGlobalStoreArguments(final String sourceName,
...
             throw new TopologyException("Processor " + processorName + " is already added.");
         }
-        if (stateFactories.containsKey(storeName) || globalStateBuilders.containsKey(storeName)) {
-            throw new TopologyException("StateStore " + storeName + " is already added.");
+        if (stateFactories.containsKey(storeName)) {
+            throw new TopologyException("A different StateStore has already been added with the name " + storeName);
+        }
+        if (globalStateBuilders.containsKey(storeName)) {
+            throw new TopologyException("A different GlobalStateStore has already been added with the name " + storeName);
         }
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java:688]: Minor improvement for the error message

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 2]
```diff
@@ -62,7 +63,7 @@
 
     private final Serde<String> stringSerde = Serdes.String();
     private final InternalTopologyBuilder builder = new InternalTopologyBuilder();
-    private final StoreBuilder<?> storeBuilder = new MockKeyValueStoreBuilder("store", false);
+    private final StoreBuilder<?> storeBuilder = new MockKeyValueStoreBuilder("testStore", false);
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:66]: Just renaming the store to make the error message in the tests easier to read.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 3]
```diff
@@ -372,48 +373,112 @@ public void testAddStateStoreWithSink() {
 
     @Test
     public void shouldNotAllowToAddStoresWithSameName() {
+        final StoreBuilder<KeyValueStore<Object, Object>> otherBuilder =
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:376]: Just refactoring:
 - add missing generic types
 - use `assertThrows`
 - add check for error message

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 4]
```diff
@@ -372,48 +373,112 @@ public void testAddStateStoreWithSink() {
...
+
+        assertThat(
+            exception.getMessage(),
+            equalTo("Invalid topology: A different StateStore has already been added with the name testStore")
+        );
     }
 
     @Test
     public void shouldNotAllowToAddStoresWithSameNameWhenFirstStoreIsGlobal() {
-        final StoreBuilder storeBuilder = new MockKeyValueStoreBuilder("store", false).withLoggingDisabled();
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:385]: As above

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 5]
```diff
@@ -372,48 +373,112 @@ public void testAddStateStoreWithSink() {
...
+        );
+
+        assertThat(
+            exception.getMessage(),
+            equalTo("Invalid topology: A different GlobalStateStore has already been added with the name testStore")
+        );
+    }
+
+    @Test
+    public void shouldNotAllowToAddStoresWithSameNameWhenSecondStoreIsGlobal() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:420]: As above (the diff is just weird... this is an existing test; cf. below)

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java [hunk 6]
```diff
@@ -372,48 +373,112 @@ public void testAddStateStoreWithSink() {
...
-                    "global-store",
-                    null,
-                    null,
-                    null,
-                    "global-topic",
-                    "global-processor",
-                    new MockProcessorSupplier<>());
-            fail("Should throw TopologyException with store name conflict");
-        } catch (final TopologyException expected) { /* ok */ }
+    public void shouldNotAllowToAddGlobalStoresWithSameName() {
```
Comment (mjsax) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java:447]: This is the missing test case.

**Review by guozhangwang — COMMENTED** [review]
The PR looks good to me. Also cc @cadonna to review as well

**Review by vvcephei — APPROVED** [review]
Thanks, @mjsax !

**Comment by mjsax** [comment]
Merged to `trunk` and cherry-picked to `2.6` branch.

**Pull request merged**